.scripts/build.ps1

```diff
@@ -114,6 +114,33 @@ function Invoke-Make([string] $makeCmd, [string] $makeFlags, [string] $sdkPath,
     $global:LASTEXITCODE = $process.ExitCode
 }
 
+# This function verifies that all project files in the mod subdirectories actually exist in the .x2proj file
+function ValidateProjectFiles([string] $modProjectRoot, [string] $modName)
+{
+    Write-Host "Checking for missing entries in .x2proj file..."
+    $projFilepath = "$modProjectRoot\$modName.x2proj"
+    if(Test-Path $projFilepath)
+    {
+        $missingFiles = New-Object System.Collections.Generic.List[System.Object]
+        # Loop through all files in subdirectories and fail the build if any filenames are missing
+        Get-ChildItem $modProjectRoot -Directory | Get-ChildItem -File -Recurse |
+        ForEach-Object {
+            If (!(Get-Content $projFilepath | Select-String -Pattern $_.Name)) {
```

Comment:
Can we move `Get-Content` out of the loop? If I'm understanding this correctly, we read the entire project file for every file we are checking.